Added Ajira Pay Finance TVL based on AJP Staking Pool on the KAVA Network

dickensodera opened the pull request:
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. The protocol is usually listed within 24 hours of merging the PR
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama): 
Ajira Pay Finance


##### Twitter Link: 
https://twitter.com/ajiraPayDefi

##### List of audit links if any: 
https://github.com/Coinsult/solidity/blob/main/Coinsult_AjiraPay_0x9D...f997_Audit.pdf


##### Website Link: 
https://portal.ajirapay.finance/


##### Logo (High resolution, will be shown with rounded borders):
https://ibb.co/Z1w8Kq0

##### Current TVL:


##### Treasury Addresses (if the protocol has treasury)
0xb00375686741591FeB3541e1740E75FE21CD9f31

##### Chain: 
kava


##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama):
Ajira Pay Finance is a Multichain Decentralized Web3 protocol for secure and seamless crypto payments.

##### Token address and ticker if any: 
0x9DBC0Ad09184226313FbDe094E7c3DD75c94f997
AJP

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Payments

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):


##### forkedFrom (Does your project originate from another project):


##### methodology (what is being counted as tvl, how is tvl being calculated):
Ajira Pay Finance Tvl Calculations are based on AJP Staking pool

g1nt0ki commented:
@dickensodera PR looks good, but soory we cant list a project with 0 tvl.

dickensodera commented:
> @dickensodera PR looks good, but soory we cant list a project with 0 tvl.

Okay thanks man. We'll check on this, do you have an idea of how to create the factories to help us retrieve the tvl data from the liquidity pools?

dickensodera commented:
> > @dickensodera PR looks good, but soory we cant list a project with 0 tvl.
> 
> Okay thanks man. We'll check on this, do you have an idea of how to create the factories to help us retrieve the tvl data from the liquidity pools?

Let me make it a draft in the meantime

g1nt0ki commented:
> > @dickensodera PR looks good, but soory we cant list a project with 0 tvl.
> 
> Okay thanks man. We'll check on this, do you have an idea of how to create the factories to help us retrieve the tvl data from the liquidity pools?

what do you mean, like uniswap factory?

dickensodera commented:
> > > @dickensodera PR looks good, but soory we cant list a project with 0 tvl.
> > 
> > 
> > Okay thanks man. We'll check on this, do you have an idea of how to create the factories to help us retrieve the tvl data from the liquidity pools?
> 
> what do you mean, like uniswap factory?

Yeah sure, we typically have the contract deployed on KAVA EMV-Co-Chain so a factory to help us fetch the liquidity pool pairs on the adapter, looking through the uniswap v2 factory codebase and docs for this

g1nt0ki commented:
you can do it like this: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/purple-bridge-dex/index.js

dickensodera commented:
> you can do it like this: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/purple-bridge-dex/index.js

Yeah sure, thanks man.

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    arbitrum                  0
    bsc                       3.7537499999999997
    ethereum                  0
    polygon                   0.1741116055877415
    kava                      0
    kava-staking              0
    staking                   0
    
    total                    3.9278616055877413

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    arbitrum                  12.001335034293827
    bsc                       13.447643720000002
    polygon                   0.34588735020989714
    ethereum                  3.5265781223468546
    kava                      3.87313
    kava-staking              0
    staking                   0
    
    total                    33.194574226850584

dickensodera commented:
Hello @g1nt0ki , could you check this?

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    arbitrum                  11.919431519424956
    ethereum                  3.502510871257329
    polygon                   0.3385745651815071
    bsc                       13.47346
    kava                      3.849315
    kava-staking              0
    staking                   0
    
    total                    33.083291955863785

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    bsc                       13.46
    polygon                   0.3405587339233167
    arbitrum                  11.902920810872024
    kava                      3.851735
    kava-staking              0
    ethereum                  3.4976592190535674
    staking                   0
    
    total                    33.052873763848915

g1nt0ki commented:
@dickensodera we cant count any of these as tvl, all I am seeing are counting tokens on wallets, for treasury metric, they need to be on multisigs and to be counted as tvl, they need to be on smart contracts and tokens deposited by users.

dickensodera commented:
> @dickensodera we cant count any of these as tvl, all I am seeing are counting tokens on wallets, for treasury metric, they need to be on multisigs and to be counted as tvl, they need to be on smart contracts and tokens deposited by users.

Hello @g1nt0ki , thanks for the response yeah. 

Maybe we missed the point of the multisigs for the treasuries, however, from the AJP token, the treasuries indicated receive the transaction fees from swaps and transfers on Arbitrum One, Polygon and Binance Smart Chain except for the KAVA token that does not charge fees/taxes.

The are the official treasury addresses for the protocol

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    kava                      0
    arbitrum                  10.620297
    bsc                       15.748124
    polygon                   0.178295816
    kava-staking              0
    staking                   0
    
    total                    26.546716816

dickensodera commented:
Hello @g1nt0ki, I have updated the treasuries to multisig from gnosis. 

I also updated the token contract treasuries to reflect the gnosis safe multisig treasuries in the below transactions.

https://bscscan.com/tx/0xbcd849670676a24329747ed2ae5d43048bb562689ab649c60717a06d1cbc563c

https://arbiscan.io/tx/0x855c0ef7b6ecbd78d7af9596c834839c74d87ad7c65b9348cd394e25bb2c23a4

Gnosis still does not support Kava, so we'll just calculate the kava tvl based on the staking statistics

g1nt0ki commented:
@dickensodera thanks for the change, we can track this as your treasury metric but your tvl would still show up as 0.

dickensodera commented:
> @dickensodera thanks for the change, we can track this as your treasury metric but your tvl would still show up as 0.

Glad to hear from you yeah.

We'll appreciate your assistance with the PR, I thought treasury balances are also counted as TVL, the tokens in a protocol's treasury that is, obtained from protocol fees  for our case as an example.

It's okay yeah, the zero tvl will change of course, it's the very beginning for us but we have a lot ahead of us.

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    arbitrum                  10.623318000000001
    polygon                   0.178115312
    bsc                       15.726867000000002
    kava-staking              0
    kava                      3.856365
    staking                   0
    
    total                    30.384665312000003

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    kava                      0
    polygon                   0.178292872
    bsc                       15.713130999999999
    kava-staking              0
    arbitrum                  10.617732
    staking                   0
    
    total                    26.509155872

dickensodera commented:
Hello @g1nt0ki , since gnosis does not yet support Kava at https://app.safe.global for the treasuries that we'd want to convert to multisigs, can we add the previous KAVA treasury that is a normal address on the Kava EVM Co-Chain for the protocol? It should basically be holding the protocol's staking tokens from the staking smart contract here https://explorer.kava.io/address/0xE6B06E3cBfDFe6AAc5e18C2eE1bFF53dFD276B39/read-contract

I removed it earlier awaiting your confirmation before merging the PR. Please advice  further so that I can push the final changes from my end.

dickensodera commented:
Hello @g1nt0ki, any updates on whether this will go through? We typically need back to the Kava Rise Program team with an update on our listing on Defillama.

g1nt0ki commented:
> Hello @g1nt0ki, any updates on whether this will go through? We typically need to get back to the Kava Rise Program team with an update on our listing on Defillama for them to take the next steps. This process is very crucial as it will determine our future on the grant program.

the issue with your PR right now is, not lack of safe support in kava, we can make an exception and count on normal address there, but it would be towards treasury, it wont show up as tvl, all the tvl components in your PR would have to moved to treasury adapter

dickensodera commented:
> > Hello @g1nt0ki, any updates on whether this will go through? We typically need to get back to the Kava Rise Program team with an update on our listing on Defillama for them to take the next steps. This process is very crucial as it will determine our future on the grant program.
> 
> the issue with your PR right now is, not lack of safe support in kava, we can make an exception and count on normal address there, but it would be towards treasury, it wont show up as tvl, all the tvl components in your PR would have to moved to treasury adapter

Thanks for the response @g1nt0ki , that's okay if everything will be moved to the treasury, let me then push the final changes and let you know for the next steps.

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    kava-staking              0
    polygon                   0.179600008
    bsc                       15.70975
    arbitrum                  10.92519
    kava                      3.6701699999999997
    staking                   0
    
    total                    30.484710008

dickensodera commented:
Hello @g1nt0ki , the above are the changes I was yet to push. You can now proceed with the next steps. Fell free to let me know if you need any further input from me. Thanks

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    bsc-staking               0
    kava-staking              0
    polygon                   0.17970599199999998
    arbitrum                  10.938756
    bsc                       15.725142000000002
    kava                      3.675885
    staking                   0
    
    total                    30.519488992000003

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    bsc-staking               0
    kava-staking              0
    arbitrum                  10.907805000000002
    bsc                       15.706852
    polygon                   0.17946348
    kava                      3.6576750000000002
    staking                   0
    
    total                    30.45179548

g1nt0ki commented:
for the last time, we cant count treasury as tvl, currently your tvl is 0, since the staking value is also 0, the listing would show all metrics except treasury as 0, is that what you want? My understanding is, kava rise has a minimum tvl requirement (check with them)

dickensodera commented:
> for the last time, we cant count treasury as tvl, currently your tvl is 0, since the staking value is also 0, the listing would show all metrics except treasury as 0, is that what you want? My understanding is, kava rise has a minimum tvl requirement (check with them)

Yeah sure @g1nt0ki , you can just proceed because the statistics will improve with time, it's just the beginning for us.

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    kava                      0
    bsc                       0
    polygon                   0
    arbitrum                  0
    bsc-staking               0
    kava-staking              0
    staking                   0
    
    total                    0

llamatester commented:
The adapter at projects/treasury/ajira-pay-finance.js exports TVL: 
        
 
 
    arbitrum-ownTokens        0
    polygon-ownTokens         0
    bsc-ownTokens             0
    arbitrum                  10.904385
    bsc                       15.702504999999999
    polygon                   0.178924728
    kava-ownTokens            0
    kava                      3.65714
    ownTokens                 0
    
    total                    30.442954727999997

llamatester commented:
The adapter at projects/ajira-pay-finance exports TVL: 
        
 
 
    kava                      0
    bsc                       0
    polygon                   0
    arbitrum                  0
    bsc-staking               0
    polygon-pool2             0
    bsc-pool2                 0
    arbitrum-pool2            0
    kava-staking              0
    staking                   0
    pool2                     0
    
    total                    0

llamatester commented:
The adapter at projects/treasury/ajira-pay-finance.js exports TVL: 
        
 
 
    polygon-ownTokens         0
    arbitrum-ownTokens        0
    bsc-ownTokens             0
    kava-ownTokens            0
    arbitrum                  10.886886
    polygon                   0.179411592
    bsc                       15.703678
    kava                      3.677335
    ownTokens                 0
    
    total                    30.447310592

dickensodera commented:
Thank you @g1nt0ki
